std/math.d

```diff
@@ -2532,74 +2532,161 @@ unittest
  *      $(TR $(TD $(NAN))            $(TD FP_ILOGBNAN) $(TD no))
  *      )
  */
-int ilogb(real x)  @trusted nothrow @nogc
+int ilogb(T)(const T x) @trusted nothrow @nogc
+    if(isFloatingPoint!T)
 {
-    version (Win64_DMD_InlineAsm)
+    import core.bitop : bsr;
+    // Provide a bsr implementation for ulong in 32-bit mode
+    int bsr_ulong(ulong x) @trusted pure nothrow @nogc
```

Comment:
There's also a ``bsr64`` implementation in ``std.numeric`` - it's private, however...  https://github.com/D-Programming-Language/phobos/blob/master/std/numeric.d#L85